p_soc_auto_base/utils.py

```diff
@@ -685,25 +686,13 @@ def get_subscription(subscription):
     :returns: a :class:`ssl_cert_tracker.models.Subscription` instance
 
     :arg str subscription: the subscription value
-        Note that this value is case-sensitive
 
-    .. todo::
-
-        Use `filter(subscription__iexact=subscription).get()` to avoid the case
-        sensitive requirement
-
-    :raises: :exc:`Exception` if the
-        :class:`ssl_cert_tracker.models.Subscription` instance cannot be found
-
-    .. todo::
-
-        Change the error catching to use a
-        :exc:`djang.db.core.exceptions.ObjectDoesNotExist` exception.
+    :raises: a :exc:`django.Model.DoesNotExist` exception if the model doesn't
+    exist.
     """
     try:
-        return Subscription.objects.\
-            get(subscription=subscription)
-    except Exception as error:
+        return Subscription.objects.get(subscription__iexact=subscription)
+    except DoesNotExist as error:
         raise error
```

Comment:
please log the error and make sure that the error message includes the value of the subscription argument